design/specs/specs-null-values.md

```diff
@@ -0,0 +1,124 @@
+As for the process of transforming values to and from NULL, it will be important to have a consistent way of doing this across different data types.
+
+## Scenarios
+
+### Scenario 1: User identifies a NULL Value
+
+#### Scenario 1a: The user identifies a NULL value in a table
+
+#### Steps for 1a
+
+1. The user opens a table that contains NULL values
+2. The user identifies a cell that contains a NULL value
+    - The cell contains the word NULL in all uppercase letters
+    - The cell has a lighter background-color
+    - The cell has a muted text color
+    - The cell has a different font style (italic)
+
+#### Scenario 1b: The user identifies a NULL value in a record preview
+
+#### Steps for 1b
+
+1. The user edits a value in a column that links to records from another table
+2. The user sees the list of records displayed in the record selection dropdown
+3. The user identifies a record that contains a NULL value
+    - The field contains the word NULL in all uppercase letters
+    - The text has a muted text color
+    - The text has a different font style (italic)
+
+### Scenario 2: User edits a NULL Value
+
+#### Scenario 2a: The data type is text
+
+#### Steps for 2a
+
+1. The user edits a NULL value in a column that is set to text data type
+2. The user double-clicks the value cell and enters edit mode
+3. The user enters a value and leaves edit mode by hitting enter or clicking outside of the active cell
+4. The NULL value is replaced by the new value
+
+#### Scenario 2b: The data type is text, and the value is empty
+
+#### Steps for 2b
+
+1. The user edits a NULL value in a column that is set to text data type
+2. The user double-clicks the value cell and enters edit mode
+3. The user doesn't enter a value and leaves edit mode by hitting enter or clicking outside of the active cell
+4. The NULL value is replaced by an empty string
+
+#### Scenario 2c: The data type is boolean with checkbox display enabled
+
+#### Steps for 2c
+
+1. The user edits a NULL value in a column that is set to boolean type with the checkbox display property enabled
+    - The field contains a checkbox in an indeterminate state
+2. The user clicks on the checkbox, going from indeterminate (NULL) to checked (TRUE).
+3. The user can click on the checkbox again to mark it as false. However, it does not go back to NULL.
+
+#### Scenario 2d: The data type is number
+
+#### Steps for 2d
+
+1. The user edits a NULL value in a column that is set to number type
+2. The user double-clicks the cell to enter edit mode
+3. The user exists in edit mode without making any changes
+4. The NULL value remains
+
+### Scenario 3: User sets a value to NULL
+
+#### Scenario 3a: The column accepts NULL values
+
+#### Steps for 3a
+
+1. The user selects a cell in a column that contains a value
+2. The user clears the content of the selected cell
+    - By pressing the delete key
+    - By opening the contextual menu and selecting the 'Set as NULL' option
```

Comment:
I'm assuming this is in "normal/navigation mode" (i.e., not "edit mode").  Is that correct?  Is there a clear way for the user to see visually which mode they're in?  Many spreadsheet programs make a distinction between these two modes for cells, and that gives different behaviors for, e.g., the delete key depending on mode.  I'd be a bit grumpy (as a user) if I hit the delete key _in edit mode_, and the whole cell blanked out (as opposed to just the character to the right of the cursor).

On that front, if they're in "normal/navigation mode", would the backspace key have the similar "set to null" action as the delete key?  It seems like that would make sense if the two modes are clearly differentiated.

Reply:
@mathemancer you are right, this is for navigation mode. In edit mode keys will have the default behavior